echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/cdevents/CDEventsSenderService.java

```diff
@@ -0,0 +1,82 @@
+/*
+    Copyright (C) 2023 Nordix Foundation.
+    For a full list of individual contributors, please see the commit history.
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+          http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    SPDX-License-Identifier: Apache-2.0
+*/
+package com.netflix.spinnaker.echo.cdevents;
+
+import io.cloudevents.CloudEvent;
+import io.cloudevents.core.message.MessageWriter;
+import io.cloudevents.http.HttpMessageFactory;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CDEventsSenderService {
+
+  /**
+   * Sends CDEvent to the Configured ClouEvent broker URL.
+   *
+   * @param ceToSend
+   * @param url
+   * @return HttpURLConnection
+   * @throws IOException
+   */
+  public HttpURLConnection sendCDEvent(CloudEvent ceToSend, URL url) throws IOException {
+    HttpURLConnection httpUrlConnection = createConnection(url);
+    MessageWriter messageWriter = createMessageWriter(httpUrlConnection);
```

Comment:
A concern here is this is REALLY raw java layer for HTTP communications.  The standard in spinnaker is to use retrofit & ok-http client libs.  These automatically add things LIKE timeouts, URL validation, etc. that are pretty key settings for most environments.  I've not looked at the CloudEvents MessageWriter system... so may HAVE to do this, but definitely have concerns.

Reply:
I am trying to use Retrofit client to send CDEvent but facing difficulties in deserializing as Retrofit doesn't directly use `CloudEventHttpMessageConverter` is the issue.
Can I use spring RestTemplate to send the CDEvent instead, please share your ideas?

Reply:
With retrofit and a return type of Response (e.g. [here](https://github.com/spinnaker/echo/blob/31779124727a5eda13224a3ba9082c2c215996af/echo-core/src/main/java/com/netflix/spinnaker/echo/services/IgorService.java#L58)), I believe you'll get flexibility on converting the response.

Reply:
I could able to solve this issue by creating a custom [CDEventsHTTPMessageConverter](https://github.com/spinnaker/echo/pull/1295/files#diff-03fe23e811c27266271bbe77d1b5879778999b8aabf512271b53cf7c5430341e) as retrofit doesn't directly support CloudEventHttpMessageConverter, this converter uses Jackson converter to map the CloudEvent as String and send this String with the Requestbody